src/looklike.ml

```diff
@@ -0,0 +1,150 @@
+(** Looklike allows you to find recurrent patterns in some lists *)
+
+
+(** Module type which allows comparing elements *)
+module type EqType = 
+sig
+  type t
+  val equal : t -> t -> bool
+  val nil : t 
+  val get_str : t -> string
+end
+
+
+
+(** OCaml type to get pattern and its position *)
+type 'a index_string = {
+  depot : 'a list;
+  pos_doc :int ;
+  pos_src :int 
+}
+
+
+
+(** Module for looklike to find similarity between list *)
+module Make (E : EqType) =
```

Comment:
Here I would write:
```ocaml
module Make (E : sig 
  include Comparable
  include Printable with type t := t
end)
```
with Comparable defined as
```
module type Comparable = sig
  type t
  val compare : t -> t -> int
end
module type Printable = sig
 type t
 val to_string : t -> string
end
```
(I still do not have a clue about the meaning of `nil`.)

Reply:
With this new version, I don't even need Printable... I can only use equal or compare.